app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,43 @@
 class PostsController < ApplicationController
   def index
     @users = User.all
-    @posts = Post.all
-    @comments = Comment.all
   end
 
   def show
     @post = Post.find(params[:id])
     @comments = Comment.all
   end
+
+  def new
+    @user = current_user
+    @new_post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    if @post.save
+      flash[:notice] = 'Post successfully added!'
+      redirect_to user_path(current_user)
+    else
+      flash[:alert] = 'Post could not be added.'
+      redirect_to new_user_post_path(current_user)
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @like = Like.new(post: @post, author: current_user)
+
+    if @like.save
+      flash[:notice] = 'You have liked the post!.'
+    else
+      flash[:alert] = "Something is wrong!, you don't liked the post!."
+    end
+
+    redirect_to user_post_path(current_user.id, @post.id)
+  end
```

Comment:
-  You have done great so far, :+1:  but there are a few changes you need to make before moving to the next milestone. 

- [ ] I noticed that the "create like" method is currently located in the PostController. While this might work, it's a better practice to place it in the LikesController for the following reasons:

**Single Responsibility Principle**: Controllers should have a single responsibility. The PostController is responsible for managing posts, and having the "create like" logic there would make the controller less focused and harder to maintain. So, I recommend moving the "create like" method to the `LikesController.` :+1: :100:

Reply:
@V-Blaze Thanks for your review!!, yeah your right, I'll change it :1st_place_medal: